Test/Tasty/ExpectedFailure.hs

```diff
@@ -1,19 +1,53 @@
-{-# LANGUAGE DeriveDataTypeable, ScopedTypeVariables #-}
+{-# LANGUAGE DeriveDataTypeable, ScopedTypeVariables, LambdaCase #-}
 module Test.Tasty.ExpectedFailure (expectFail, expectFailBecause, ignoreTest, ignoreTestBecause, wrapTest) where
 
 import Test.Tasty.Options
 import Test.Tasty.Runners
 import Test.Tasty.Providers
+import Test.Tasty ( Timeout(..) )
 import Data.Typeable
 import Data.Tagged
 import Data.Maybe
 import Data.Monoid
+import Control.Exception ( displayException, try, SomeException )
+import Control.Concurrent.Timeout ( timeout )
+
 
 data WrappedTest t = WrappedTest (IO Result -> IO Result) t
     deriving Typeable
 
 instance forall t. IsTest t => IsTest (WrappedTest t) where
-    run opts (WrappedTest wrap t) prog = wrap (run opts t prog)
+    run opts (WrappedTest wrap t) prog =
+      -- re-implement timeouts and exception handling *inside* the wrapper
```

Comment:
You say “re-implement”. does this mean that the following code is a copy from somewhere else (e.g. tasty)? If so, can you link to that place, so that it’s easier to check for and copy improvements to that code?

Reply:
Yes it was in the original tasty code.  I've updated the comment to reference that.